modules/apps/transfer/types/trace.go

```diff
@@ -76,15 +67,23 @@ func (dt DenomTrace) GetFullDenomPath() string {
 	return dt.GetPrefix() + dt.BaseDenom
 }
 
-// extractPathAndBaseFromFullDenom returns the trace path and the base denom from
-// the elements that constitute the complete denom.
-func extractPathAndBaseFromFullDenom(fullDenomItems []string) ([]string, string) {
+// ExtractDenomFromFullPath returns the denom from the full path.
+// Used to support v1 denoms.
+func ExtractDenomFromFullPath(fullPath string) Denom {
```

Comment:
this should be the only function we need to go from: path -> Denom

Reply:
Maybe `FullPathToDenom` ([ref](https://github.com/cosmos/ibc-go/pull/6426#discussion_r1619414270)) or if we move this function to `denom.go` then maybe `FromFullPath` or `NewDenom`.

Reply:
going to leave as is unless there's consensus otherwise